addons/l10n_latam_invoice_document/wizards/account_debit_note.py

```diff
@@ -5,12 +5,16 @@ class AccountDebitNote(models.TransientModel):
 
     _inherit = 'account.debit.note'
 
-    def create_debit(self):
-        """ Properly compute the latam document type of type debit note. """
-        res = super().create_debit()
-        new_move_id = res.get('res_id')
-        if new_move_id:
-            new_move = self.env['account.move'].browse(new_move_id)
-            new_move._compute_l10n_latam_document_type()
-            new_move._onchange_l10n_latam_document_type_id()
-        return res
+    def _prepare_default_values(self, move):
+        default_values = super()._prepare_default_values(move)
+
+        # properly compute debit type
+        debit_note = self.env['account.move'].new({
+            'move_type': default_values.get('move_type'),
+            'journal_id': default_values.get('journal_id'),
+            'partner_id': move.partner_id.id,
+            'company_id': move.company_id.id,
+        })
+        document_types = debit_note.l10n_latam_available_document_type_ids.filtered(lambda x: x.internal_type == 'debit_note')
+        default_values['l10n_latam_document_type_id'] = document_types and document_types[0].ids[0]
+        return default_values
```

Comment:
Hello @zaoral, the `create_debit` override you removed already manually recomputes the document type. So that sets the appropriate document type on the debit note in the end. It makes the document type that's being copied sort of irrelevant I think. So to get around the constrains error can we not copy it at all?

```suggestion
    def _prepare_default_values(self, move):
        vals = super()._prepare_default_values(move)
        vals['l10n_latam_document_type_id'] = False
        return vals
```

If we do that and keep the existing `create_debit` override I think it should take care of your issue. It's also a fairly minimal change that should be safer for a stable release.

We also should add a test. `l10n_latam_invoice_document` has none at the moment, but you can take inspiration from another small module. Maybe [l10n_latam_check](l10n_latam_document_type_id), but feel free to look around in our `l10n_*` modules. I'm happy to help you with that, just let me know!

cc @jco-odoo I know you were looking at this as well, let me know if you have another idea.